exercises/practice/protein-translation/tests/Tests.elm

```diff
@@ -0,0 +1,178 @@
+            \() ->
+                ProteinTranslation.proteins "UAA"
+                    |> Expect.equal (Ok [])
+        , -- skip <|
+          test "STOP codon RNA sequence 2" <|
+            \() ->
+                ProteinTranslation.proteins "UAG"
+                    |> Expect.equal (Ok [])
+        , -- skip <|
+          test "STOP codon RNA sequence 3" <|
+            \() ->
+                ProteinTranslation.proteins "UGA"
+                    |> Expect.equal (Ok [])
+        , -- skip <|
+          test "Sequence of two protein codons translates into proteins" <|
+            \() ->
+                ProteinTranslation.proteins "UUUUUU"
+                    |> Expect.equal (Ok [ "Phenylalanine", "Phenylalanine" ])
+        , -- The codons are different, the proteins they translate into are not.
+          -- skip <|
+          test "Sequence of two different protein codons translates into proteins" <|
+            \() ->
+                ProteinTranslation.proteins "UUAUUG"
+                    |> Expect.equal (Ok [ "Leucine", "Leucine" ])
+        , -- skip <|
+          test "Translate RNA strand into correct protein list" <|
+            \() ->
+                ProteinTranslation.proteins "AUGUUUUGG"
+                    |> Expect.equal (Ok [ "Methionine", "Phenylalanine", "Tryptophan" ])
+        , -- skip <|
+          test "Translation stops if STOP codon at beginning of sequence" <|
+            \() ->
+                ProteinTranslation.proteins "UAGUGG"
+                    |> Expect.equal (Ok [])
+        , -- skip <|
+          test "Translation stops if STOP codon at end of two-codon sequence" <|
+            \() ->
+                ProteinTranslation.proteins "UGGUAG"
+                    |> Expect.equal (Ok [ "Tryptophan" ])
+        , -- skip <|
+          test "Translation stops if STOP codon at end of three-codon sequence" <|
+            \() ->
+                ProteinTranslation.proteins "AUGUUUUAA"
+                    |> Expect.equal (Ok [ "Methionine", "Phenylalanine" ])
+        , -- skip <|
+          test "Translation stops if STOP codon in middle of three-codon sequence" <|
+            \() ->
+                ProteinTranslation.proteins "UGGUAGUGG"
+                    |> Expect.equal (Ok [ "Tryptophan" ])
+        , -- skip <|
+          test "Translation stops if STOP codon in middle of six-codon sequence" <|
+            \() ->
+                ProteinTranslation.proteins "UGGUGUUAUUAAUGGUUU"
+                    |> Expect.equal (Ok [ "Tryptophan", "Cysteine", "Tyrosine" ])
+        , -- It's up to a track how to behave when a non-existing codon is
+          -- passed; to simplify the exercise, this test may be excluded in
+          -- order to remove the requirement for input validation.
+          -- skip <|
+          test "Non-existing codon can't translate" <|
+            \() ->
+                ProteinTranslation.proteins "AAA"
+                    |> Expect.equal (Err "Invalid codon")
```

Comment:
Let's use types all the way, I think it's the most idiomatic way.
```suggestion
                    |> Expect.equal (Err InvalidCodon)
```

Reply:
I've made the other edits, but I'll need to work my way through adding the types and chasing out the ramifications. I haven't touched Elm in a while so I'm a little rusty. I can poke at it more this weekend.